packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -508,9 +597,9 @@ export const createDateStrForInputFromSections = (sections: FieldSection[]) => {
   return `\u2066${formattedArray.join('')}\u2069`;
 };
 
-export const getSectionsBoundaries = <TDate, TSection extends FieldSection>(
+export const getSectionsBoundaries = <TDate>(
```

Comment:
I could not pass the full section anymore because of the digit editing on letter section (and the other way around).
So I'm just passing the `format` and `contentType`.